src/iuse_actor.cpp

```diff
@@ -4289,3 +4290,199 @@ iuse_actor *weigh_self_actor::clone() const
+    // Returns true if the item already has the mod or if we have enough materials and thread to add it
+    const auto can_add_mod = [&]( const std::string & new_mod, const itype_id & mat_item ) {
+        return mod.item_tags.count( new_mod ) > 0 ||
+               ( it.charges >= thread_needed && has_enough[mat_item] );
+    };
+
+    uilist tmenu;
+    // TODO: Tell how much thread will we use
+    if( it.charges >= thread_needed ) {
+        tmenu.text = _( "How do you want to modify it?" );
+    } else {
+        tmenu.text = _( "Not enough thread to modify. Which modification do you want to remove?" );
+    }
+
+    int index = 0;
+    for( auto cm : clothing_mods ) {
+        // TODO: List how much material we have and how much we need
+        auto obj = cm.obj();
+        item temp_item = modded_copy( mod, obj.flag );
+        temp_item.update_clothing_mod_val();
+        bool enab = can_add_mod( obj.flag, obj.item_string );
+        std::string prompt;
+        if( mod.item_tags.count( obj.flag ) == 0 ) {
+            prompt = obj.implement_prompt;
+        } else {
+            prompt = obj.destroy_prompt;
+        }
+        std::ostringstream desc;
+        desc << string_format( _( "Bash/Cut: %d/%d->%d/%d" ),
+                               mod.bash_resist(), mod.cut_resist(),
+                               temp_item.bash_resist(), temp_item.cut_resist() ) << "\n";
+        desc << string_format( _( "Acid: %d->%d" ),
+                               mod.acid_resist(), temp_item.acid_resist() ) << "\n";
+        desc << string_format( _( "Fire: %d->%d" ),
+                               mod.fire_resist(), temp_item.fire_resist() ) << "\n";
+        desc << string_format( _( "Warmth: %d->%d" ),
+                               mod.get_warmth(), temp_item.get_warmth() ) << "\n";
+        desc << string_format( _( "Encumbrance: %d->%d" ),
+                               mod.get_encumber( p ), temp_item.get_encumber( p ) ) << "\n";
+        desc << string_format( _( "Storage: %s %s->%s %s" ),
+                               format_volume( mod.get_storage() ), volume_units_abbr(),
+                               format_volume( temp_item.get_storage() ), volume_units_abbr() ) << "\n";
+        tmenu.addentry_desc( index++, enab, MENU_AUTOASSIGN, string_format( "%s", _( prompt.c_str() ) ),
+                             desc.str() );
+    }
+    tmenu.textwidth = 80;
+    tmenu.desc_enabled = true;
+    tmenu.query();
+    const int choice = tmenu.ret;
+
+    if( choice < 0 || choice >= static_cast<int>( clothing_mods.size() ) ) {
+        return 0;
+    }
+
+    // The mod player picked
+    const std::string &the_mod = clothing_mods[choice].obj().flag;
+
+    // If the picked mod already exists, player wants to destroy it
+    if( mod.item_tags.count( the_mod ) ) {
+        if( query_yn( _( "Are you sure?  You will not gain any materials back." ) ) ) {
+            mod.item_tags.erase( the_mod );
```

Comment:
I believe you're missing a `mod.update_clothing_mod_val();` call here to correctly update the clothing.

Reply:
fixed.